freqtrade/optimize/hyperopt.py

```diff
@@ -153,8 +155,19 @@ def log_results(self, results) -> None:
         Log results if it is better than any previous evaluation
         """
         print_all = self.config.get('print_all', False)
-        if print_all or results['loss'] < self.current_best_loss:
+        is_best_loss = results['loss'] < self.current_best_loss
+        if print_all or is_best_loss:
+            if is_best_loss:
+                self.current_best_loss = results['loss']
             log_str = self.format_results_logstring(results)
+            # Colorize output
+            if self.config.get('print_colorized', False):
+                if results['total_profit'] > 0:
+                    log_str = Style.BRIGHT + log_str
+                if results['loss'] >= MAX_LOSS:
+                    log_str = Fore.RED + log_str
+                elif is_best_loss:
+                    log_str = Fore.GREEN + log_str
```

Comment:
From looking at these results (see screenshot below) - i'm not sure if the meaning of `BRIGHT` and `GREEN` shouldn't be reversed (green when it's profitable, Bright when it's the current best)?

8/20 seems more important in the screenshot - however the best result is 12/20 (yes, i'm using the default loss function - but that's not the point).

![hyperopt_loss_color](https://user-images.githubusercontent.com/5024695/62823305-1562ec00-bb90-11e9-94d8-147fdf174b9b.png)

another thing i found strange is 1/20 and 3/20 - they have negative profit (-> lose my money) - but are green too. i would color these red too ... so it's clearer?

Reply:
Let's adjust the colorization schema if needed.

The lines in red -- are so called 'bad' results (those with objective equal or higher than MAX_LOSS value). I.e. the epochs with zero trades or with number of trades that are less than the number set with `--min-trades`. They can also be profitable. So the bold attribute (which can be used in conjunction with red color) was choosen. We cannot use green and red in the same time...

So the current colorization schema is the following:
* 'bad' epochs -- red (profitable 'bad's -- bold red). Not seen in no-print-all mode.
* current best -- green (profitable bests -- bold green). In no-print-all all results are green since hyperopt prints only bests.
* others -- white (default color, more precisely) (profitable others -- bold white(bold default color))
* the resulting best result is not colorized since it's not needed.

Any idea on how to change the schema particularly?

Reply:
how about
* current best: yellow (or any other color apart from blue) (profitable best - green)
* bad epoch - red
* others: white

It's mostly the distinction between green and bold green which is not very clear i think

Reply:
need to view this on a white terminal with black default foreground...

Reply:
that's why i said "or any other color and not blue (blue is often too dark on black backgrounds - mainly on windows putty... )

maybe orange then ... ?

Reply:
how about just inversing green and bright?

so "positive" is green - and current best is bold?

Reply:
with yellow it's impossible to follow -- it's like a rainbow, too many colors
you may see, I committed this version...

Reply:
I tagged the schema above ^^ as `colorized-schema-1`

Also committed the next one (tagged as `colorized-schema-2`): green for profitable epochs, red for losses, bright for bests (can be both red and green), dimmed for bad objectives (also both red and green). There is no ANSI terminal sequence for italic, so dimmed color was used for bads. On Windows colorama renders dimmed as normal, you won't see dimmed colors. It still looks like a new year tree (6 different colors), but at least eyes can navigate in it, in my opinion...

I also have a next variant...

Reply:
Here we are, colorization schema-3 (tagged as `colorized-schema-3`): green for profits, red for losses, bright for bests, but only green ones. We actually do not quite need to highlight bad epochs and bests with losses -- they still are simply losses (simply red) or profits which are not best (simply green)...

I think my initial schema was pretty fine (I tagged it as `colorized-schema-0`) -- it contained lots of white lines so the green and red (and also bold) were not so blind for eyes... This last one is also, ghm, let's say fine... At least its can be explained better in the docs...

Reply:
colorized-schema-2 (dimmed) is too dark i think
colorized-schema-1 seems okish to me

colorized-schema-3 is "too" colorfull i guess
so 0 or 1 should win - i'll leave this to you ... thanks for giving us choice! :+1:

Reply:
Well, for me colorized-schema-1 (with yellow) is too hard to follow with eyes, colorized-schema-2 (dimmed) is also too colorized (many diff. tones of colors).

So as for above, I vote for colorized-schema-0 and colorized-schema-3...

However, I created `colorized-schema-4` (also tagged it), pretty simple: green for profitable, bold(bright) for bests. Bads are not highlighted, because 1) losses are still losses, 2) profits are still profits, so even epochs that were detected as bads may also be of interest for the user. So, no red color in this schema, let's reserve red for something more important in the output.

Huh?

Reply:
4 wins :+1: 

You're right, loss is loss (the reason is not that important ...)

Reply:
I could squash all intermediate commits with previous schemas into the final colorization schema, if this would make sence...

Reply:
yes, please

Reply:
done.